### ISSUE TITLE
[ENH]: smaller deployment install

### ISSUE THREAD

**Issue by astrobenhart** [pycaret/pycaret#3492]
### Describe the feature you want to add to this project

Hi All, 

Love developing models using PyCaret, so much of my time has been saved. I do have a feature request though, when deploying in Docker, the image is huge (bigger than it needs to be) due to package requirements in PyCaret that aren't used in prod. I have sort of gotten around this by creating a fork and cutting out sections of code and requirements but it is difficult to maintain and feels a bit like a house of cards.

Could we please have a prod/deployment version?

### Describe your proposed solution

A version of PyCart that doesn't come with any plotting, exploration tools, or models. An extra slim install that can only run load_model and predict_model for a given pipeline trained in PyCaret.

### Describe alternatives you've considered, if relevant

Forking and cutting out sections of code. Hard to maintain, not sure if will fail at some point.

**Comment by ngupta23** [comment]
Are you using the latest 3.0.0 release? It should ideally have a much lower dependency footprint than the 2.x release.

**Comment by astrobenhart** [comment]
No, I'm not. I'm using 2.3.10. I developed some models using this version so I'm kind of locked in to using it in prod. I'll rebuild a model using 3.0.0 and take a look. @ngupta23 , do you have an estimated size of the image using the latest version?

**Comment by ngupta23** [comment]
No, I don't. I just know that it should have a much smaller dependency. Also, since the NLP module was removed in 3.0.0, all the big language dictionaries that were needed earlier in 2.x would also not be needed anymore in 3.0.0.

**Comment by Rodrigolorcas** [comment]
I have been trying this for ages. Here is a workaround I found:

You can save your model without the pipeline and then load it with joblib to generate predictions (I would love to use the pipeline, but I don't know how tyo do it yet)

**Comment by moezali1** [comment]
Closing as part of the 4.0 backlog cleanup — see #4177.

PyCaret 4.0 is a ground-up rewrite (OOP-only API, sklearn-native engine, Python 3.11+, sklearn ≥ 1.7) and almost all pre-4.0 issues either reference removed APIs (`setup()`, `plot_model`, `evaluate_model`, `interpret_model`, `automl`, the legacy time-series module, the Streamlit dashboard, …) or no longer reproduce on the new code path.

**If this still affects you in 4.0, please re-file as a new issue** with:

- the experiment constructor call (`Experiment(...)`),
- the operation that fails,
- the full traceback,
- your Python / sklearn / pycaret versions.

Resources for 4.0:

- Docs: https://pycaret.org/docs
- Migrate from 3.x: https://pycaret.org/docs/guides/migrating-from-3-x
- API reference: https://pycaret.org/reference
- FAQ: https://pycaret.org/docs/resources/faqs

Thanks for filing — and apologies for the long latency. The repo had drifted; we're starting clean.
